packages/opentelemetry-sdk-trace-web/src/utils.ts

```diff
@@ -125,6 +129,129 @@ export function addSpanNetworkEvents(
   }
 }
 
+function _getBodyNonDestructively(body: ReadableStream) {
+  // can't read a ReadableStream without destroying it
+  // we CAN "tee" the body stream, which lets us split it, but that still locks the original stream
+  // so we end up needing to return one of the forks.
+  const [bodyToReturn, bodyToConsume] = body.tee();
```

Comment:
Note: tee (may) not exist in all supported runtimes (added in 2022), so there should be some code which checks for the presence of this capability to avoid failures when running in an older runtime.

Reply:
Would clone help, similar to what's done here? - https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-instrumentation-fetch/src/fetch.ts#L347

Reply:
> Would clone help, similar to what's done here?

So the `Request` object has a `clone()` method on it, which is what I use in the else block of `getFetchBodyLength`-- https://github.com/open-telemetry/opentelemetry-js/pull/4706/files#diff-52544e5ffe6a8d0257ea6d658d0aae0f70f99b650cf123392e7f6a8e6eb236bbR204-R212

Unfortunately, the `ReadableStream` object does not have a clone method on it, so if we receive that instead of a Request object, we have to take this approach.

Reply:
> Note: tee (may) not exist in all supported runtimes (added in 2022), so there should be some code which checks for the presence of this capability to avoid failures when running in an older runtime.

Thanks for calling this out @MSNev! Did you see any other APIs that might be like this? What's the best way for me to check this?

Reply:
I didn't notice anything else specifically, and mostly it's based on my decades of history that causes me to question the presence, which I then go and research the availability of the "niggling" feature by looking it up on W3C pages, ES definitions and caniuse.com